### PR TITLE
KOGITO-7606 Custom dashboards

### PR DESCRIPTION
**Pull request by bxf12315**
**JIRA:**
https://issues.redhat.com/browse/KOGITO-7606 

**Description:**

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributions doc](https://github.com/kiegroup/kogito-docs/blob/main/CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [ ] The nav.adoc file has a link to this guide in the proper category
- [ ] The index.adoc file has a card to this guide in the proper category, with a meaningful description

<details>
<summary>
How to setup automatic cherry-pick to another branch ?
</summary>

The cherry-pick action allows to setup automatic cherry-pick from `main` to a specific branch.

To allow it, you will need to add the corresponding label with pattern: `backport-{RELEASE_BRANCH}`.  
For example, if a backport to branch `1.26.x` is needed, then the label should be `backport-1.26.x`.

Once the PR is merged, the action will retrieve the commit and cherry-pick it to the desired branch.

*NOTE: You can still add label after the merge and it should still be cherry-picked.*
</details>

### CODE REVIEW

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc [hunk 1]
```diff
@@ -0,0 +1,21 @@
+= Dashboards in Kogito Serverless Workflow Tools extension
```
Comment (radtriste) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:1]: How do you access this page ? Missing into the `nav.adoc` file ?

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+In Kogito Serverless Workflow Tools, the Dashboards page is used to displays the available dashboard files. The page displays a list of available dashboard and add filters to the list.
+
+.Dashboard page
+image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-dashboard-list.png[]
+
+The table on the Dashboards page displays the following details:
+
+* *Name*: Displays the name of a custom dashboard.
+* *Last Modified*: Displays the time reference indicating when the custom dashboard is last updated.
+
```
Comment (radtriste) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:12]:
```suggestion
* *Name*: displays the name of a custom dashboard.
* *Last Modified*: displays the time reference indicating when the custom dashboard is last updated.

```

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+To filter the dashboard based on name, enter a name in the *Filter by name* field and click *Apply filter* button.
```
Comment (radtriste) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:13]: screenshot is showing `Filter by dashboard` field

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+.Example filters applied on the dashboards
+image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-filters-in-dashbaords.png[]
+
+In the dashboard table, the *name* column contains a clickable link, which enables you to navigate to the *Dashboard Details* page.
```
Comment (radtriste) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:18]: I would setup 2 subsections to this page:

- Filtering the dashboard(s)
- Accessing a dashboard details

That way we would have a right menu with the direct link to what we want to achieve. wdyt ?

**Review by radtriste — CHANGES_REQUESTED** [review]

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc (hunk 1 above)
Comment (bxf12315) [reply serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:1]: ok

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc (hunk 4 above)
Comment (bxf12315) [reply serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:20]: done

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc [hunk 5]
```diff
@@ -0,0 +1,23 @@
+= Dashboards in Kogito Serverless Workflow Tools extension
+
+In Kogito Serverless Workflow Tools, the Dashboards page is used to displays the available dashboard files. The page displays a list of available dashboard and add filters to the list.
```
Comment (paulovmr) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:3]:
```suggestion
In Kogito Serverless Workflow Tools, the Dashboards page is used to display the available dashboard files. The page displays a list of available dashboards and add filters to the list.
```

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+.Dashboard page
+image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-dashboard-list.png[]
+
+The table on the Dashboards page displays the following details:
+
+* *Name*: displays the name of a custom dashboard.
```
Comment (paulovmr) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:10]:
```suggestion
* *Name*: the name of a custom dashboard.
```

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+* *Last Modified*: displays the time reference indicating when the custom dashboard is last updated.
```
Comment (paulovmr) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:11]:
```suggestion
* *Last Modified*: the time reference indicating when the custom dashboard was last updated.
```

**Review by paulovmr — CHANGES_REQUESTED** [review]
Hi @bxf12315 , I think one important thing to add here is the project directory where the dashboards should be saved to show in the extension.

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+=== Filtering the dashboards
+To filter the dashboard based on name, enter a name in the *Filter by dashboard name* field and click *Apply filter* button.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:14]:
```suggestion
To filter the dashboard based on name, enter a name in the *Filter by dashboard name* field and click the *Apply filter* button.
```

**Review by kaldesai — APPROVED** [review]
@bxf12315 One minor suggestion, otherwise your content looks good to me. Thank you!

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc [hunk 9]
```diff
@@ -0,0 +1,23 @@
+= Dashboards in Kogito Serverless Workflow Tools extension
```
Comment (radtriste) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:1]: The page is missing those:
```
:compat-mode!:
// Metadata:
:description: ...
:keywords: kogito, workflow, serverless, Quarkus, Dev UI, dashboards
```

Comment (radtriste) [reply serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:1]: See https://raw.githubusercontent.com/kiegroup/kogito-docs/main/serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-workflow-definition-page.adoc as an example

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+.Dashboard page
+image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-dashboard-list.png[]
+
+The table on the Dashboards page displays the following details:
+
+* *Name*: displays the name of a custom dashboard.
+* *Last Modified*: displays the time reference indicating when the custom dashboard is last updated.
+
+=== Filtering the dashboards
```
Comment (radtriste) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:13]:
```suggestion
== Filtering the dashboards
```

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+To filter the dashboard based on name, enter a name in the *Filter by dashboard name* field and click *Apply filter* button.
+
+.Example filters applied on the dashboards
+image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-filters-in-dashbaords.png[]
+
+=== Accessing a dashboard details
```
Comment (radtriste) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:19]:
```suggestion
== Accessing a dashboard details
```

**Review by radtriste — CHANGES_REQUESTED** [review]

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+
+.Dashboard page
+image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-dashboard-list.png[]
+
+The table on the Dashboards page displays the following details:
+
+* *Name*: the name of a custom dashboard.
+* *Last Modified*: the time reference indicating when the custom dashboard was last updated.
+
+== Creating custom dashboard
```
Comment (paulovmr) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:19]:
```suggestion
== Creating a custom dashboard
```

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+
+=== Create a custom dashboard file
+The {dashboard_guide}[dashboard guide] for creating dashboards and visualizations with YAML. You can run all examples with {dashboard_editor}[Dashbuilder YML Online].
+
+=== Storage path of custom dashboard
```
Comment (paulovmr) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:24]:
```suggestion
=== Storage path of custom dashboards
```

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+Default storage path is *dashboards*, user also can use property *quarkus.kogito-runtime-tools.custom.dashboard.folder* to set new custom storage path.
```
Comment (paulovmr) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:25]:
```suggestion
The default storage path for dashboard files is *src/main/resources/dashboards*, but the property *quarkus.kogito-runtime-tools.custom.dashboard.folder* can be used to set a custom storage path.
```

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+
+.Default storage path
+image::tooling/quarkus-dev-ui-extension/kogtio-swf-tools-dashboard-storage-path.png[]
```
Comment (paulovmr) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:28]:
```suggestion
image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-dashboard-storage-path.png[]
```
The file should be renamed too.

Comment (bxf12315) [reply serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc]: Done

**Review by ricardozanini — APPROVED** [review]

**Review by paulovmr — APPROVED** [review]

**Comment by ricardozanini** [comment]
Moving to DP4. @radtriste are you ok with this PR?

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+To filter the dashboard based on name, enter a name in the *Filter by dashboard name* field and click *Apply filter* button.
+
+.Example filters applied on the dashboards
+image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-filters-in-dashbaords.png[]
+
+=== Accessing a dashboard details
+In the dashboard table, the *name* column contains a clickable link, which enables you to navigate to the *Dashboard Details* page.
+
+.Dashboard Details page
+image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-dashboard-details.png[]
```
Comment (radtriste) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:40]: Missing the `Found an issue` part. See other pages as the last section

Comment (bxf12315) [reply serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:40]: Done

File: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+
+The table on the Dashboards page displays the following details:
+
+* *Name*: the name of a custom dashboard.
+* *Last Modified*: the time reference indicating when the custom dashboard was last updated.
+
+== Creating a custom dashboard
+
+=== Create a custom dashboard file
+The {dashboard_guide}[dashboard guide] for creating dashboards and visualizations with YAML. You can run all examples with {dashboard_editor}[Dashbuilder YML Online].
```
Comment (MarianMacik) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:22]: This sentence is missing something, it sounds more like a description.
`The {dashboard_guide}[dashboard guide] for creating dashboards and visualizations with YAML.`

Comment (MarianMacik) [inline serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc:22]: We should probably unify YAML/YML, so mostlikely YAML.
```suggestion
The {dashboard_guide}[dashboard guide] for creating dashboards and visualizations with YAML. You can run all examples with {dashboard_editor}[Dashbuilder YAML Online].
```

Comment (bxf12315) [reply serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc]: Done

Comment (MarianMacik) [reply serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc]: This still holds. What is the purpose of this sentence? `The {dashboard_guide}[dashboard guide] for creating dashboards and visualizations with YAML.` Does it describe any picture or something similar?

Comment (bxf12315) [reply serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc]: Custom dashboard has to create a dashboard file first, so we need let user know how to create it. 
Create dashboard does not belong to rumtime-tools, so I think the external reference is better.

Comment (MarianMacik) [reply serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc]: I know, just the sentence is weirdly written to me :) It preferably could be `See the dashboard guide for creating dashboards and visualizations with YAML.` or similar.

Comment (bxf12315) [reply serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc]: Done

**Review by MarianMacik — APPROVED** [review]

**Review by radtriste — APPROVED** [review]

**Comment by radtriste** [comment]
sorry Marian took over that one so I was not looking.
One QE approval should be enough ;)

**Comment by ricardozanini** [comment]
@domhanak you wanna take a look to? :)

**Review by domhanak — APPROVED** [review]

**Comment by ricardozanini** [comment]
@bxf12315 @radtriste do we need to backport this?

**Comment by radtriste** [comment]
backport to ?

**Comment by ricardozanini** [comment]
> 

to another branch

**Comment by MarianMacik** [comment]
@ricardozanini I think we could backport it to 1.34.

**Pull request merged**